python/common/python/math/__init__.java

```diff
@@ -0,0 +1,840 @@
+        return new org.python.types.Float(Math.abs(toFloatValue(v)));
+    }
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"a1", "a2"}
+    )
+    public static org.python.Object fmod(org.python.Object a1, org.python.Object a2) {
+        double v = toFloatValue(a1);
+        double w = toFloatValue(a2);
+        if (isnan(v) || isnan(w)) {
+            return new org.python.types.Float(Double.NaN);
+        }
+        if (isinf(w)) {
+            return a1;
+        }
+        if (w == 0.0) {
+            throw new org.python.exceptions.ValueError("math domain error");
+        }
+        if (isinf(v) && w == 0.0) {
+            throw new org.python.exceptions.ValueError("math domain error");
+        }
+        return new org.python.types.Float(v % w);
+    }
+
+
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"v"}
+    )
+    public static org.python.Object exp(org.python.Object v) {
+        double r = Math.exp(toFloatValue(v));
+        if (exceptInf(r, toFloatValue(v))) {
+            throw new org.python.exceptions.OverflowError("math range error");
+        } else {
+            return new org.python.types.Float(r);
+        }
+    }
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"v"}
+    )
+    public static org.python.Object floor(org.python.Object v) {
+        if (v instanceof org.python.types.Int || v instanceof org.python.types.Float) {
+            return new org.python.types.Int((int) Math.floor(toFloatValue(v)));
+        } else {
+            try {
+                org.python.Object[] args = new org.python.Object[1];
+                java.util.Map<java.lang.String, org.python.Object> kwargs = new java.util.HashMap<java.lang.String, org.python.Object>();
+                org.python.Object oper = v.__getattribute__("__floor__");
+                return (org.python.Object) ((org.python.types.Method) oper).invoke(args, kwargs);
+            } catch (org.python.exceptions.AttributeError e) {
+                if (org.Python.VERSION < 0x03060000) {
+                    throw new org.python.exceptions.TypeError("a float is required");
+                } else {
+                    throw new org.python.exceptions.TypeError("must be real number, not " + v.typeName());
+                }
+            }
+        }
+    }
+
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"v"},
+            default_args = {"base"}
+    )
+    public static org.python.Object log(org.python.Object v, org.python.Object base) {
+        double b = 0.0;
+        if (base instanceof org.python.types.Float || base instanceof org.python.types.Int) {
+            b = toFloatValue(base);
+        } else if (base == null) {
+            b = 0.0;
+        }
+        double doubleValue;
+        doubleValue = log(toFloatValue(v));
+        return (b == 0.0) ? new org.python.types.Float(doubleValue) : new org.python.types.Float(applyLoggedBase(doubleValue, b));
```

Comment:
Could the conditional be moved inside the `Float()` constructor args? Similar to line 203 above